File: network_bfs.cpp
#include <vector>
#include <queue>
#define MAX 200
using namespace std;

int solution(int n, vector<vector<int>> computers) {
    int answer = 0, node;
    int visited[MAX] = {0};
    queue<int> q;
    
    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            if(computers[i][j] && !visited[i]) {
                answer++;
                q.push(i);
                while(!q.empty()) {
                    node = q.front(); q.pop();
                    if(!visited[node]) {
                        visited[node] = 1;
                        for(int k=0; k<n; k++)
                            if(computers[node][k] && !visited[k])
                                q.push(k);
                    }
                }
            }
        }
    }
    
    return answer;
}

File: network_bfs.py
def solution(n, computers):
    answer = 0
    visited = [0 for _ in range(n)]
    queue = []
    
    for i in range(n):
        for j in range(n):
            if computers[i][j] and not visited[i]:
                answer += 1
                queue.append(i)
                
                while len(queue):
                    node = queue[0]; del queue[0]
                    if not visited[node]:
                        visited[node] = 1
                        for k in range(n):
                            if computers[node][k] and not visited[k]:
                                queue.append(k)

    return answer